_rules/css-orientation-lock-media-queries-b33eff.md

```diff
@@ -0,0 +1,209 @@
+---
+id: b33eff
+name: Page has no orientation lock specified using CSS media queries
+rule_type: atomic
+description: |
+  This rule checks that page content is not locked to any single display orientation using CSS media queries.
+accessibility_requirements:
+ wcag20:1.3.4: # Orientation
+  forConformance: true
+  failed: not satisfied
+  passed: further testing needed
+  inapplicable: further testing needed
+input_aspects:
+- DOM Tree
+- CSS Styling
+authors:
+- Jey Nandakumar
+- Audrey Maniez
+---
+
+## Applicability
+
+The rule applies to any element that is [visible](#visible) and has a CSS [transform](https://developer.mozilla.org/en-US/docs/Web/CSS/transform) property with any of the below [transform-functions](https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function)
+
+- [rotate transform-function](https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function/rotate)
+- [rotate3d transform-function](https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function/rotate3d)
+- [rotateX transform-function](https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function/rotateX)
+- [rotateY transform-function](https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function/rotateY)
+- [rotateZ transform-function](https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function/rotateZ)
+- [matrix transform-function](https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function/matrix)
+- [matrix3d transform-function](https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function/matrix3d)
+
+that are applied conditionally on the [orientation](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/orientation) [media feature](https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Using_media_queries#Media_features) with a value of `landsapce` or `portrait`.
+
+## Expectation
+
+The target element has a [current transformation matrix](https://drafts.csswg.org/css-transforms/#current-transformation-matrix) that does not restrict the element to only `landspace` or `portrait` orientation.
```

Comment:
I'd maybe consider a slightly more abstract formulation:

> The target element is not rotated around the Z axis at an angle corresponding to +-90 degrees relative to the position of the element prior to transformation.

One way to determine the rotation of an element would of course be to look at the current transformation matrix, but I do think that's an implementation detail. 

Maybe @Jym77 has some input as well?

Reply:
Once again, adding `rotation` makes it ambiguous. And adding a number +-90deg is implementation specific I would argue (Also 180/ 360 is allowed rotation values), so again very implementation specific.

I prefer to leave this as is, because the linked definition of current transformation matrix, in terms looks into coordinate systems, which should detail all things that are axis related.

Reply:
I disagree on all accounts. Let's continue the discussion here: https://github.com/act-rules/act-rules.github.io/pull/408#discussion_r318907852